internal/runtime/client/client_test.go

```diff
@@ -699,6 +752,32 @@ func TestClient_CallExtension(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name:                       "should succeed with correct NamespaceSelector",
+			registeredExtensionConfigs: []runtimev1.ExtensionConfig{correctNamespaceSelector},
+			testServer: testServerConfig{
+				start: false,
+			}, args: args{
+				hook:     fakev1alpha1.FakeHook,
+				name:     "valid-extension",
+				request:  &fakev1alpha1.FakeRequest{},
+				response: &fakev1alpha1.FakeResponse{},
+			},
+			wantErr: false,
+		},
+		{
+			name:                       "should succeed with incorrect NamespaceSelector",
```

Comment:
Testing needs to be done to make sure the extension is being called or not. Probably easier in a seperate unit test here.